lambeq/text2diagram/model_based_reader/base.py

```diff
@@ -0,0 +1,196 @@
+# Copyright 2021-2024 Cambridge Quantum Computing Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Model-based reader
+==================
+Base class for readers that use pre-trained models, either end-to-end
```

Comment:
I think "Base class for readers that use pre-trained models for generating diagrams." should be enough here?